hoodie-client/src/test/java/com/uber/hoodie/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -581,6 +587,171 @@ public void testCommitWritesRelativePaths() throws Exception {
     }
   }
 
+  @Test
+  public void testUpsertsForMultipleRecordsInSameFile() throws Exception {
```

Comment:
Can we just create a IOHandle level test for this? I would like to leave this file for API level tests.